blocktracker/blocktracker.go

```
@@ -382,31 +370,31 @@ func NewSubscriptionBlockTracker(logger *log.Logger, client *jsonrpc.Client) (*S
 // Track implements the BlockTracker interface
 func (s *SubscriptionBlockTracker) Track(ctx context.Context, handle func(block *ethgo.Block) error) error {
 	data := make(chan []byte)
+	defer close(data)
+
 	cancel, err := s.client.Subscribe("newHeads", func(b []byte) {
 		data <- b
 	})
 	if err != nil {
 		return err
 	}
+	defer cancel()
 
-	go func() {
-		for {
-			select {
-			case buf := <-data:
-				var block ethgo.Block
-				if err := block.UnmarshalJSON(buf); err != nil {
-					s.logger.Printf("[ERR]: Tracker failed to parse ethgo.Block: %v", err)
-				} else {
-					handle(&block)
```

This error was being ignored completely